$OpenBSD: patch-t_protocol_version_t,v 1.1 2021/09/01 15:04:37 bluhm Exp $

LibreSSL does not define TLS1_3_VERSION.  So disable TLSv1_3 by
setting version to TLSv1_2.

Index: t/protocol_version.t
--- t/protocol_version.t.orig
+++ t/protocol_version.t
@@ -99,6 +99,10 @@ if ($pid == 0) {
 	    last if $_ eq $ver;
 	    $sslver .= ":!$_";
 	}
+	# LibreSSL can disable TLSv1_3 only by setting to TLSv1_2
+	if (Net::SSLeay::constant("LIBRESSL_VERSION_NUMBER")) {
+	    $sslver = "TLSv1_2" if $ver eq "TLSv1_2";
+	}
 	$check->($sslver,$ver);
     }
 
